app/src/main/java/com/adevinta/mappicker/MainActivity.kt

```diff
@@ -71,50 +74,63 @@ class MainActivity : AppCompatActivity() {
             MainView()
         }
 
-        initializeLocationPickerTracker()
-    }
-
-    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
-        super.onActivityResult(requestCode, resultCode, data)
-        if (resultCode == Activity.RESULT_OK && data != null) {
-            Log.d("RESULT****", "OK")
-            if (requestCode == 1) {
-                val latitude = data.getDoubleExtra(LATITUDE, 0.0)
-                Log.d("LATITUDE****", latitude.toString())
-                val longitude = data.getDoubleExtra(LONGITUDE, 0.0)
-                Log.d("LONGITUDE****", longitude.toString())
-                val address = data.getStringExtra(LOCATION_ADDRESS)
-                Log.d("ADDRESS****", address.toString())
-                val postalcode = data.getStringExtra(ZIPCODE)
-                Log.d("POSTALCODE****", postalcode.toString())
-                val bundle = data.getBundleExtra(TRANSITION_BUNDLE)
-                Log.d("BUNDLE TEXT****", bundle?.getString("test").toString())
-                val fullAddress = data.getParcelableExtra<Address>(ADDRESS)
-                if (fullAddress != null) {
-                    Log.d("FULL ADDRESS****", fullAddress.toString())
-                }
-                val timeZoneId = data.getStringExtra(TIME_ZONE_ID)
-                if (timeZoneId != null) {
-                    Log.d("TIME ZONE ID****", timeZoneId)
+        lekuActivityResultLauncher =
+            registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result: ActivityResult ->
+                if (result.resultCode == Activity.RESULT_OK) {
+                    onResult(result.resultCode, result.data)
+                } else {
+                    Log.d("RESULT****", "CANCELLED")
```

Comment:
ActivityResultLaunchers are usually initialized on the property definition, so you can avoid using `late init var` and use a `val`.